src/coreclr/src/jit/emit.cpp

```diff
@@ -5320,50 +5328,40 @@ UNATIVE_OFFSET emitter::emitFindOffset(insGroup* ig, unsigned insNum)
  *  block.
  */
 
-UNATIVE_OFFSET emitter::emitDataGenBeg(UNATIVE_OFFSET size, bool align)
+UNATIVE_OFFSET emitter::emitDataGenBeg(UNATIVE_OFFSET size, UNATIVE_OFFSET alignment)
 {
     unsigned     secOffs;
     dataSection* secDesc;
 
     assert(emitDataSecCur == nullptr);
 
-    /* The size better not be some kind of an odd thing */
+    // The size must not be zero and must be a multiple of 4
+    assert((size != 0) && ((size % 4) == 0));
 
-    assert(size && size % sizeof(int) == 0);
+    // This restricts the alignment to: 1, 2, 4, 8, 16, or 32
+    // Alignments greater than 32 would require VM support in ICorJitInfo::allocMem
 
-    /* Get hold of the current offset */
+    const size_t MaxAlignment = 32;
+    assert(isPow2(alignment) && (alignment <= MaxAlignment));
 
+    /* Get hold of the current offset */
     secOffs = emitConsDsc.dsdOffs;
 
-    if (align)
+    if (alignment > 4)
```

Comment:
Some comments here would be useful. It's not entirely clear what's special about 4 here.

Reply:
Added the following:
> // As per the above comment, the minimum alignment is actually 4
        // bytes so we don't need to make any adjustments if the requested
        // alignment is 1, 2, or 4.
        //
        // The maximum requested alignment is tracked and the memory allocator
        // will end up ensuring offset 0 is at an address matching that
        // alignment. So if the requested alignment is greater than 4, we need
        // to pad the space out so the offset is a multiple of the requested.